scripts/iso-manager.sh

```diff
@@ -139,14 +80,29 @@ _set_env() {
     fi
 }
 
-# helper function to check for element in array
-containsElement () {
-  local element match="$1"
-  shift
-  for element in "$@"; do
-      [[ "$element" == "$match" ]] && return 0;
-  done
-  return 1
+get_configured_archives() {
+    salt-call pillar.get metalk8s:archives --out yaml |
+        sed -rn 's/^- (.+)$/\1/p'
+}
+
+_check_archives() {
+    local -a not_existing
+
+    for archive; do
+        if [ ! -e "$archive" ]; then
+            not_existing+=("$archive")
+        fi
+    done
+
+    if (( ${#not_existing[@]} )); then
+        echo "The following archives do not exist:" >&2
+        for archive in "${not_existing[@]}"; do
+            echo "- $archive" >&2
+        done
+        return 1
+    fi
+
+    return 0
 }
```

Comment:
IMHO this check should be in salt so that we can also check this in bootstrap, also this remind me this ticket #2862

Reply:
I agree, but having it here allow to fail early before touching this file.
Maybe we need it here and in Salt too or we could also do like we did for Solutions, not using file.serialize and having a custom module adding/removing ISO from this file.

Reply:
The salt module could take archive list as input (and default to pillar content)

Hmm yes maybe it make sense to have a dedicated Salt module to add iso to bootstrap file